### DISCUSSION TITLE
[@types/react] false types for HTMLInput attribute checked

### DISCUSSION THREAD

**Discussion by maxkarkowski** [DefinitelyTyped/DefinitelyTyped#70760] [Issues with a @types package] [answered]
Hi there,
i stumbled upon an issue regarding the checked attributes in HTMLInput attributes
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/2234e7a3407a8cf765e05951d923dae4848bb664/types/react/index.d.ts#L3362

According to the docs and specs true and false are not valid optiones for the html attribute checked,
https://html.spec.whatwg.org/multipage/common-microsyntaxes.html#boolean-attributes

Quote:

> The values "true" and "false" are not allowed on boolean attributes. To represent a false value, the attribute has to be omitted altogether.

So if an input is checked(true) it should have the attribute set 
example:
if checked is true
```html
<input type="radio" checked />
<!-- or --> 
<input type="radio" checked="checked" />
<!-- or -->
<input type="radio" checked="" />
```

if an input is not checked the only valid way is to not set the checked attribute completely
```html
<input type="radio" /> <!-- not checked -->
```

[a comment by an automated account is not shown]

**Comment by eps1lon** [comment, marked as answer]
These are the types for React that not necessarily map 1:1 to the HTML attributes. `<input type="radio" checked={false} />` is valid in React.

**Reply by maxkarkowski** [reply]
Yeah i get that. but would a be not better to be consistent with the html specification? using the types in libs like radix could cause to a faulty state of an input. i will look more into it especially in my case using radix. 
so i can totally understand your answer :)

**Reply by eps1lon** [reply]
> but would a be not better to be consistent with the html specification?

React does not match HTML types since we're writing JS not HTML

>  using the types in libs like radix could cause to a faulty state of an input

Radix-UI is using React. If they use HTML types, that's a bug in their types you should report.
